src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeField.cs

```diff
@@ -24,16 +24,19 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
         public readonly DeclarationScope Scope;
 
+        public readonly ConstantValue? DefaultValue;
+
         /// <summary>Anonymous type field type</summary>
         public TypeSymbol Type => TypeWithAnnotations.Type;
 
-        public AnonymousTypeField(string name, Location location, TypeWithAnnotations typeWithAnnotations, RefKind refKind, DeclarationScope scope)
+        public AnonymousTypeField(string name, Location location, TypeWithAnnotations typeWithAnnotations, RefKind refKind, DeclarationScope scope, ConstantValue? defaultValue = null)
```

Comment:
Any IDE concerns regarding https://github.com/dotnet/roslyn/blob/c18a66e5912ba8ad6183eed6d2496ae8221a52d0/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.AnonymousTypeSymbolKey.cs#L65-L67 ? `CreateAnonymousTypeSymbol` doesn't pass `defaultValue` when creating `AnonymousTypeField`

cc @CyrusNajmabadi

Reply:
We should file a prototype comment to track this and sync with IDE team before main merge

Reply:
Just added